scripts/dependabot_generate_changesets.ts

```diff
@@ -0,0 +1,93 @@
+import fsp from 'fs/promises';
+import { EOL } from 'os';
+import { GitClient } from './components/git_client.js';
+import { readPackageJson } from './components/package-json/package_json.js';
+
+const createChangesetFile = async (
+  versionUpdates: Map<string, string>,
+  packageNames: string[]
+) => {
+  let message = '';
+
+  for (const [dep, version] of versionUpdates) {
+    message += `Update dependency ${dep} to ${version}.${EOL}`;
+  }
+
+  const frontmatterContent = packageNames
+    .map((name) => `'${name}': patch`)
+    .join(EOL);
+  const body = `---${EOL}${frontmatterContent}${EOL}---${EOL}${EOL}${message.trim()}${EOL}`;
+  await fsp.writeFile(fileName, body);
+};
+
+const getVersionUpdates = async (files: string[]) => {
+  const updates = new Map<string, string>();
+  for (const file of files) {
+    const changes = await gitClient.getFileChanges(file);
+    for (const change of changes.split(EOL)) {
+      if (!change.startsWith('+ ')) {
+        continue;
+      }
+      // This will capture lines in git diff like:
+      // +  "<dependency>": "^<version>"
+      const match = change.match(/"(.*)": "(.*)"/);
+
+      if (!match) {
+        continue;
+      }
+      // Add dependency name (match[1]) and new version (match[2]) to Map
+      updates.set(match[1].replace(/"/g, ''), match[2].replace(/"/g, ''));
+    }
+  }
+
+  return updates;
+};
```

Comment:
One other source to consider is parse PR title or description. I'm not sure if this will be any simple - please evaluate and adopt or reject this suggestion.

Reply:
Sometimes dependabot will group version updates together so the title can be something like `Bump the npm_and_yarn group across 1 directory with 3 updates`.

I think description will be best, since it gives us the version dependabot updated from since there will be a line like `Updates/Bumps <dependency> from <old-version> to <new-version>`.